pallets/gear/src/benchmarking/syscalls.rs

```diff
@@ -262,6 +262,37 @@ where
         Self::prepare_handle(module, 0)
     }
 
+    pub fn free_range(repetitions: u32, pages_per_call: u32) -> Result<Exec<T>, &'static str> {
+        use Instruction::*;
+
+        let n_pages = repetitions.checked_mul(pages_per_call).unwrap();
+        assert!(n_pages <= max_pages::<T>() as u32);
+
+        let mut instructions = vec![];
+        for _ in 0..API_BENCHMARK_BATCH_SIZE {
+            instructions.extend([I32Const(n_pages as i32), Call(0), I32Const(-1)]);
+            unreachable_condition(&mut instructions, I32Eq); // if alloc returns -1 then it's error
+
+            for i in 0..repetitions {
```

Comment:
```suggestion
        for _ in 0..repetitions {
            instructions.extend([I32Const(n_pages as i32), Call(0), I32Const(-1)]);
            unreachable_condition(&mut instructions, I32Eq); // if alloc returns -1 then it's error

            for i in 0..API_BENCHMARK_BATCH_SIZE {
```

Reply:
repetitions can be zero so this could potentially break time estimations in benchmarks (because no alloc would happen)

Reply:
Consider
```rust
for _ in 0..repetitions.max(1) {
```

Reply:
It is expected behavior to exclude overhead on module instantiation and etc